File: python/src/models/compass/core/query_run.py
from datetime import datetime
from typing import Any, Dict, List, Optional

from pydantic import BaseModel

from .tags import Tags


class QueryRun(BaseModel):
    id: str
    sqlStatementId: str
    state: str
    path: str
    fileCount: Optional[int] = None
    lastFileNumber: Optional[int] = None
    fileNames: Optional[str] = None
    errorName: Optional[str] = None
    errorMessage: Optional[str] = None
    errorData: Optional[Any] = None
    dataSourceQueryId: Optional[str] = None
    dataSourceSessionId: Optional[str] = None
    startedAt: Optional[str] = None
    queryRunningEndedAt: Optional[str] = None
    queryStreamingEndedAt: Optional[str] = None
    endedAt: Optional[str] = None
    rowCount: Optional[int] = None
    totalSize: Optional[int] = None
    tags: Tags
    dataSourceId: str
    userId: str
    createdAt: str
    updatedAt: datetime
    archivedAt: Optional[datetime] = None

File: js/src/types/compass/core/query-run.ts
import { Tags } from "./tags";

export interface QueryRun {
  id: string;
  sqlStatementId: string;
  state: string;
  path: string;
  fileCount: number | null;
  lastFileNumber: number | null;
  fileNames: string | null;
  errorName: string | null;
  errorMessage: string | null;
  errorData: any | null;
  dataSourceQueryId: string | null;
  dataSourceSessionId: string | null;
  startedAt: string | null;
  queryRunningEndedAt: string | null;
  queryStreamingEndedAt: string | null;
  endedAt: string | null;
  rowCount: number | null;
  totalSize: number | null;
  tags: Tags;
  dataSourceId: string;
  userId: string;
  createdAt: string;
  updatedAt: string; // Assuming that datetime translates to a string in TypeScript
  archivedAt: string | null; // Assuming that datetime translates to a string in TypeScript
}
